test/javascripts/acceptance/canned-replies-test.js

```diff
@@ -1,139 +1,76 @@
-            excerpt: "This is an example canned reply",
-            content:
-              "This is an example canned reply.\nYou can user **markdown** to style your replies. Click the **new** button to create new replies or the **edit** button to edit or remove an existing canned reply.\n\n*This canned reply will be added when the replies list is empty.*",
-          },
-          {
-            id: "1a1987620aa49135344abe65eb43302d",
-            title: "Testing",
-            excerpt: "",
-            content: "<script>alert('ahah')</script>",
-            usages: 1,
-          },
-          {
-            id: "5317b7fd066d3d4c15acde92d70f0377",
-            title: "This is a test",
-            excerpt: "Testing",
-            content: "Testing testin **123**",
-            usages: 1,
-          },
-          {
-            id: "04697870e02acfef3c2130dab92fe6d8",
-            title: "Using variables",
-            excerpt:
-              "Hi %{reply_to_username,fallback:there}, regards %{my_username}.",
-            content:
-              "Hi %{reply_to_username,fallback:there}, regards %{my_username}.",
-          },
-        ],
-      });
-    });
+  needs.user({
+    can_use_canned_replies: true,
   });
+
+  needs.pretender(cannedRepliesPretender);
   needs.hooks.beforeEach(() => clearPopupMenuOptionsCallback());
 
-  test("Inserting canned replies", async (assert) => {
+  test("Filtering by tags", async (assert) => {
+    const popUpMenu = await selectKit(".toolbar-popup-menu-options");
+
     await visit("/");
 
     await click("#create-topic");
-    await fillIn(".d-editor-input", "beforeafter");
-
-    const editorInput = $(".d-editor-input")[0];
-    editorInput.selectionStart = editorInput.selectionEnd = "before".length;
-
-    const popUpMenu = await selectKit(".toolbar-popup-menu-options");
     await popUpMenu.expand();
     await popUpMenu.selectRowByValue("showCannedRepliesButton");
-    await click(".canned-reply-title");
 
-    assert.ok(
-      find(".canned-replies-content")
-        .html()
-        .indexOf("<strong>markdown</strong>") !== -1,
-      "it should display the right cooked content"
-    );
+    const tagDropdown = selectKit(".canned-replies-filter-bar .tag-drop");
+    await tagDropdown.expand();
 
-    await click(
-      "#canned-reply-ce5fc200ab90dd0d5ac597ca9bb4708b .canned-replies-apply"
+    await tagDropdown.fillInFilter(
+      "cupcake",
+      ".canned-replies-filter-bar .tag-drop input"
+    );
+    assert.deepEqual(
+      tagDropdown.displayedContent(),
+      [
+        {
+          name: "cupcakes",
+          id: "cupcakes",
+        },
+      ],
+      "it should filter tags in the dropdown"
     );
 
+    await tagDropdown.selectRowByIndex(0);
     assert.equal(
-      find(".d-editor-input").val(),
-      "before\n\n**markdown**\n\nafter",
-      "it should contain the right selected output"
+      document.querySelectorAll(".canned-replies-list .canned-reply").length,
```

Comment:
`count(selector)` helper from `discourse/tests/helpers/qunit-helpers` could be used there 😃 

(the same applies to the test below)

Reply:
`count(selector)` uses queryAll that uses jQuery in:

https://github.com/discourse/discourse/blob/98f3349c314463f69f26a207459ac21eb78bfeae/app/assets/javascripts/discourse/tests/helpers/qunit-helpers.js#L463

Since you're deprecating that it is better to leave the code as is or use `count` anyway?

Reply:
`count` still preferred 😃 The implementation will be de-jQueryfied at some point 😃